### PR TITLE
Some fixes and improvements done while testing

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: cli/cli.go [hunk 1]
```diff
@@ -344,17 +344,15 @@ OPTIONS:
 								"logs from " +
 								"ERROR level",
 						},
+						cli.StringFlag{
```
Comment (alfrunes) [inline cli/cli.go:347]: Hmm... I would propose moving all these flags up one level to the main command scope as they will probably become relevant for the upcoming "upload-to-server" sub-command.

File: cli/snapshot_test.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+//    limitations under the License.
+package cli
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWrongCompressionArgs(t *testing.T) {
```
Comment (alfrunes) [inline cli/snapshot_test.go:22]: :+1:

**Review by alfrunes — COMMENTED** [review]
@kacf What do you think about the flag scopes? Should we keep them at the sub-command level even though they might become relevant for other sub-commands in the future?

**Comment by ghost** [comment]
> What do you think about the flag scopes? Should we keep them at the sub-command level even though they might become relevant for other sub-commands in the future?

Semantically I can see the reasoning why we should, but there are some practical reasons for why I think we should not:

* Discoverability: It's hard to discover which flags you can use when they are not documented together with the command you're using
* Scriptability: We had to move the `--compression` flag from a global command to the sub commands in mender-artifact, because scripts very often can only add arguments to the end. Without it it was very difficult to provide the means to change the flag as it had to be in a certain position in the command line

Also, I think that having flags between two sub commands, although not strictly invalid, has a somewhat strange feel to it.

```
mender snapshot -fs-path / dump
vs
mender snapshot dump -fs-path /
```

Somehow I'm always expecting the sub command to come first, don't you agree?

**Pull request closed without merging**